### ISSUE TITLE
Icons for tabs keep colour when swiching tab

### ISSUE THREAD

**Issue by Jacalz** [fyne-io/fyne#5310]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When selecting a tab, the text and icon switches to the primary colour and this should then revert back to previous state once a different tab is selected. The v2.5.3-rc6 release has a problem where the icon doesn't change colour when the selected tab changes.

### How to reproduce

1. Run the example code below and switch to the second tab.
2. Notice that the icon for the first tab still has the primary colour.

### Screenshots

[Screencast From 2024-12-11 14-39-40.webm](https://github.com/user-attachments/assets/3d27c28d-a532-4cd2-a87a-de3bef6b2acf)

### Example code

```go

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/theme"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello")

	tabs := container.NewAppTabs(
		container.NewTabItemWithIcon("1", theme.InfoIcon(), widget.NewEntry()),
		container.NewTabItemWithIcon("2", theme.MailSendIcon(), widget.NewEntry()),
	)

	w.SetContent(tabs)
	w.ShowAndRun()
}
```

### Fyne version

v2.5.3-rc6

### Go compiler version

1.23.3

### Operating system and version

Fedora Silverblue 41

**Comment by Jacalz** [comment]
I don't think there were changes to the tab code in this release right? If this is a result of image refresh changes then we might have similar issues elsewhere (assuming that the tab code hasn't incorrectly been not refreshing images when it should).

**Comment by andydotxyz** [comment]
Both of those statements are correct - it was caused by the image changes, but it is because we were not refreshing appropriately.
Other issues were found, and resolved, in check and radio earlier.

**Comment by andydotxyz** [comment]
Fixed in v2.5.3-rc7
